### PR TITLE
T14: Temporary fix of `statusBarColor`

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #1585

### CODE REVIEW

**Review by FabianDevel — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt [hunk 1]
```diff
@@ -183,13 +192,24 @@ abstract class TwoPaneFragment : Fragment() {
     }
 
     fun openThread(uid: String) {
+
         mainViewModel.currentThreadUid.value = uid
-        slidingPaneLayout.open()
+
+        val isOpening = slidingPaneLayout.openPane()
+
+        if (isOpening) requireActivity().window.statusBarColor = requireContext().getColor(R.color.backgroundColor)
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:200]: Shouldn't this be called when the `currentThreadUid` is observed?

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt [hunk 2]
```diff
@@ -183,13 +192,24 @@ abstract class TwoPaneFragment : Fragment() {
...
     }
 
     private fun resetPanes() {
-        slidingPaneLayout.close()
+
+        val isClosing = slidingPaneLayout.closePane()
+
+        if (isClosing && this is ThreadListFragment) {
+            requireActivity().window.statusBarColor = requireContext().getColor(R.color.backgroundHeaderColor)
+        }
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:209]: Shouldn't this be called when the `currentThreadUid` is observed as well?

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt [hunk 3]
```diff
@@ -162,7 +162,16 @@ abstract class TwoPaneFragment : Fragment() {
 
     override fun onConfigurationChanged(newConfig: Configuration) {
         super.onConfigurationChanged(newConfig)
+
         ensureThreadIsOpen()
+
+        val isLeftShown = areBothShown() || isOnlyLeftShown() // TODO: Only works on Phone, and not on Tablet.
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:168]: Currently the naming feels kind of confusing, is there no better way of naming things?

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 4]
```diff
@@ -193,9 +190,10 @@ class ThreadFragment : Fragment() {
             toolbarSubject.setTextColor(textColor)
         }
 
-        changeToolbarColorOnScroll(toolbar, messagesListNestedScrollView) { color ->
-            appBar.backgroundTintList = ColorStateList.valueOf(color)
-        }
+        // TODO: Disabled while there is no Tablet design
+        // changeToolbarColorOnScroll(toolbar, messagesListNestedScrollView) { color ->
+        //     appBar.backgroundTintList = ColorStateList.valueOf(color)
+        // }
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:196]: We shouldn't remove this, is there no workaround?

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:186]: Done in https://github.com/Infomaniak/android-kMail/pull/1606

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt (hunk 2 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:195]: eDone in https://github.com/Infomaniak/android-kMail/pull/1606

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt (hunk 3 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt:163]: I personnaly don't find it confusing

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt [hunk 5]
```diff
@@ -162,7 +162,16 @@ abstract class TwoPaneFragment : Fragment() {
...
+        val statusBarColor = if (isLeftShown && this is ThreadListFragment) {
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt]: Done.

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt (hunk 4 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:196]: I didn't find any workaround while there is no Tablet design.

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt]: Done.

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneFragment.kt]: Can't make it work for now. It seems that we need to handle activity recreation to fix that.

**Review by FabianDevel — APPROVED** [review]

**Review by LunarX — CHANGES_REQUESTED** [review]
When I open a bottom sheet (like the bottom right "more" menu) and close it inside the ThreadFragment, the status bar goes pink instead of staying white.

Also, I now remember a thing I did for the menu drawer. We can make it so the status bar color gradually updates according to the progress of the drawer opening. This will make the color shift less abrupt

File: app/src/main/java/com/infomaniak/mail/ui/main/search/SearchFragment.kt [hunk 6]
```diff
@@ -98,6 +98,7 @@ class SearchFragment : TwoPaneFragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+        requireActivity().window.statusBarColor = requireContext().getColor(R.color.backgroundColor)
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/main/search/SearchFragment.kt:101]: Why did this need to be added?

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/search/SearchFragment.kt (hunk 6 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/search/SearchFragment.kt]: Oversight when decoupling TwoPanelFragment from its children.

**Pull request merged**